### PR TITLE
Initial Module Creation

### PR DESCRIPTION
**Pull request by thathaneydude**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,44 @@
 # terraform-azure-sensor
+
 Terraform for Corelight's Azure Cloud Sensor Deployment.
+
+<img src="docs/overview.svg" alt="overview">
+
+## Usage
```
Comment (retr0h) [inline README.md:7]: Do we not want to follow the GCP sensor pattern where we link off to the example?

Comment (retr0h) [reply README.md:7]: Currently this Readme deviates from our others, style wise.

File: load_balancer.tf [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  port                = 443
+  request_path        = "/api/system/healthcheck/"
+  protocol            = "Https"
+  interval_in_seconds = 30
+  probe_threshold     = 3
+}
+
+resource "azurerm_lb_rule" "monitoring_vxlan_lb_rule" {
+  loadbalancer_id                = azurerm_lb.scale_set_lb.id
+  name                           = "vxlan-lb-rule"
```
Comment (retr0h) [inline load_balancer.tf:37]: Resource name is hard coded, should this be a var?

File: load_balancer.tf [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+resource "azurerm_lb_backend_address_pool" "monitoring_pool" {
+  loadbalancer_id = azurerm_lb.scale_set_lb.id
+  name            = "monitoring-pool"
+}
+
+resource "azurerm_lb_probe" "sensor_health_check_probe" {
+  loadbalancer_id     = azurerm_lb.scale_set_lb.id
+  name                = "health-check"
```
Comment (retr0h) [inline load_balancer.tf:27]: Resource name is hard coded, should this be a var?

File: load_balancer.tf [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+resource "azurerm_lb_backend_address_pool" "management_pool" {
+  loadbalancer_id = azurerm_lb.scale_set_lb.id
+  name            = "management-pool"
+}
+
+resource "azurerm_lb_backend_address_pool" "monitoring_pool" {
+  loadbalancer_id = azurerm_lb.scale_set_lb.id
+  name            = "monitoring-pool"
```
Comment (retr0h) [inline load_balancer.tf:22]: Resource name is hard coded, should this be a var?

File: load_balancer.tf [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  backend_port                   = 4789
+  frontend_port                  = 4789
+  frontend_ip_configuration_name = azurerm_lb.scale_set_lb.frontend_ip_configuration[0].name
+  backend_address_pool_ids = [
+    azurerm_lb_backend_address_pool.monitoring_pool.id
+  ]
+}
+
+resource "azurerm_lb_rule" "monitoring_geneve_lb_rule" {
+  name                           = "geneve-lb-rule"
```
Comment (retr0h) [inline load_balancer.tf:48]: ... should these resource names be vars?

File: README.md (hunk 1 above)
Comment (thathaneydude) [reply README.md:7]: I don't think it hurts to have a usage section and examples. 🤷

File: load_balancer.tf (hunk 2 above)
Comment (thathaneydude) [reply load_balancer.tf:37]: All the resources that are somewhat "hidden" as internal configs in other resources have hard coded names. I'll pull them out to variables with defaults though for consistency sake.

**Review by retr0h — APPROVED** [review]

**Pull request merged**